File: RA1/clase12/JAVA/Main.java
public class Main{
    public static void main(String[] args) {
        Animal a1 = new Animal("a1", "sin especie", "sin raza", -1, -1);

        a1.move();
        a1.eat();
        a1.makeSound();

        Gato garfield = new Gato("Garfield", "felino", "Persa", 40, 20, 10);
        garfield.move();
        garfield.eat();
        garfield.makeSound();
        garfield.purr();

        Perro clifford = new Perro("Clifford", "canino", "Labrador", 2, 113, "puro");

        clifford.move();
        clifford.eat();
        clifford.makeSound();
        clifford.bark();

        Erizo sonic = new Erizo("Sonic", "Insectiboro", "Erizo Orejon", 15, 35, 6);

        sonic.move();
        sonic.eat();
        sonic.makeSound();
        sonic.seHaceBolita();

        //Generalizacion del uso de la herencia
        Animal[] lista = new Animal[4];

        lista[0] = a1;
        lista[1] = garfield;
        lista[2] = clifford;
        lista[3] = sonic;

        for(Animal a : lista){
            System.out.println("El animal " + a.nombre + " es " + a.especie);
            a.makeSound();
        }
    }
}

File: RA1/clase12/PYTHON/Main.py
from Animal import Animal
from Gato import Gato
from Perro import Perro
from Erizo import Erizo

if __name__ == "__main__":

    a = Animal("","","",-1,-1)
    garfield = Gato("Garfield", "felino", "Persa", 42, 10, 10)
    clifford = Perro("Clifford", "Canino", "Labrador", 2, 113, "alto")
    Sonic = Erizo("Sonic", "Insectiboro", "Orejon", 15, 35, 6)

    garfield.sleep()
    clifford.sleep()
    Sonic.sleep()
    #a.sleep()

    garfield.eat()
    garfield.eat("Lazanna")
    """
    garfield.make_sound()
    clifford.make_sound()
    Sonic.make_sound()
    """

    lista = []
    lista = [None]*3 #Inicializacion de un arreglo vacio en python de 3 posiciones
    lista[0] = garfield
    lista[1] = clifford
    lista[2] = Sonic

    for a in lista:
        a.make_sound()